### PR TITLE
Fix password configuration error issue

### PR DESCRIPTION
**Pull request by Rex-Chen-NXP**
Fix sap mix mode psk and sae password configure error issue.

### CODE REVIEW

**Comment by Rex-Chen-NXP** [comment]
Hi @jukkar  @krish2718 @MaochenWang1 
Build and format check pass in local setup, could you help review?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -649,6 +649,8 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 		case 'p':
 			params->psk = state->optarg;
 			params->psk_length = strlen(params->psk);
+			params->sae_password = optarg;
+			params->sae_password_length = strlen(params->sae_password);
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:653]: no point in using the both,  as both have different rules, may be we should add a new option for SAE with no length restrictions.

Comment (Rex-Chen-NXP) [reply subsys/net/l2/wifi/wifi_shell.c:653]: Not using both, just set SAE password by '-p'

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:653]: I mean, setting both `psk` and `sae_password`. `-p` for `psk` and `-s` (if not already used) for `sae_password`.

Comment (MaochenWang1) [reply subsys/net/l2/wifi/wifi_shell.c:653]: Maybe we don't need to set `params->sae_password`?
currently WPA3 SAE connection works fine.

Comment (Rex-Chen-NXP) [reply subsys/net/l2/wifi/wifi_shell.c:653]: Both password for WPA2 and WPA3 are psk, even through psk for WPA2 is derive PTK, for WPA3 is get P value from EC， but for user, there is no difference, so I prepare to delete this change, and remove the 'sae_password' element in 'wifi_connect_req_params' later. Then psk used for both WPA2 and WPA3.
What do you think about this?

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:653]: The usage is same, but the length restrictions are differnet, `psk` -> `8..63` and `sae_password` is `8..` (no limit).

**Comment by Rex-Chen-NXP** [comment]
Hi @krish2718  @MaochenWang1  @jukkar 
Remove the shell configuration for sae password, will do more test and code view, then raise PR for it.
Could you help review again.

**Review by MaochenWang1 — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]
Can you extend commit log with "hostapd doesn't expect double-quoted string like wpa_supplicant"

**Pull request merged**